app/labor/badge_rewarder.rb

```diff
@@ -1,8 +1,17 @@
 module BadgeRewarder
   def self.award_yearly_club_badges
-    award_one_year_badges
-    award_two_year_badges
-    award_three_year_badges
+    years = 3
+    (1..years).each do |i|
+      message = "Happy DEV birthday! Can you believe it's been #{i} years already?!"
+      User.where("created_at < ? AND created_at > ?", i.year.ago, i.year.ago - 2.day).find_each do |user|
+        achievement = BadgeAchievement.create(
+          user_id: user.id,
+          badge_id: Badge.find_by_slug("#{i}-year-club").id,
```

Comment:
This looks like it'll run the `Badge.find_by_slug("#{i}-year-club")` query for each user found. I would probably move that query next to where `message` is defined to have it run once.

```ruby
    (1..years).each do |i|
      message = "Happy DEV birthday! Can you believe it's been #{i} years already?!"
      badge = Badge.find_by_slug("#{i}-year-club")
```

Then you reference the same badge instance for every iteration of that year.

```ruby
        achievement = BadgeAchievement.create(
          user_id: user.id,
          badge_id: badge.id,
```

Reply:
I used the already available code but indeed moving it out of the query loop would be better.

Reply:
Actually you should be able to simplify it even more and rely on Rails association handling.

```ruby
        achievement = BadgeAchievement.create(
          user: user,
          badge: badge,
```

Reply:
@abraham If I remove `.id` the tests fail.

Reply:
Hmmm...I wouldn't expect that to cause a test failure. Feel free to cleanup that minor cleanup suggestion or I can take a look later today.

Reply:
I will add it now with `.id`